src/kili/services/plugins/upload.py

```diff
@@ -183,6 +183,9 @@ def _retrieve_plugin_src(self) -> List[Path]:
 
         file_path = self.plugin_path
 
+        if not check_file_contains_handler(self.plugin_name, str(file_path)):
```

Comment:
If you add `check_file_contains_handler` here, you should remove it from `parse_script`. Otherwise, the code will get unclear since we will do it twice at two different places

Reply:
But here it is only for one file plugins, it will not go through here if multifile
I wanted to use the fact that you check if the path is a directory here, for checking each file one by one I still need the for that call parse_script

Reply:
I didn't notice this one but yes it seems to be the most effective solution, I will do that